macropad/code/keys.py

```diff
@@ -0,0 +1,197 @@
+'''
+YOU CAN EDIT THIS FILE!
+
+This is where you tell the macro pad which buttons do what.
+
+The macro pad has pages, where each page can have up to 8 buttons
+(corresponding to the 8 buttons on the macro pad).
+
+These are defined in the "pages" variable below. Each page has
+a name that will be displayed on the screen, and then a list
+of buttons. Each button has a name and an action. So in the
+example below, "Text Utilities" is the page name, and "Cut",
+"Copy" and "Paste" are the button names.
+
+>   pages = (
+>       (
+>           "Text Utilities", (
+>               ("Cut", (Keycode.CONTROL, Keycode.X)),
+>               ("Copy", (Keycode.CONTROL, Keycode.C)),
+>               ("Paste", (Keycode.CONTROL, Keycode.V)),
+>           )
+>       ),
+>   )
+
+You'll need to be really careful of using the right number of
+brackets and commas, otherwise the program won't know what is
+a button or a page.
+
+The first button defined in the list will be the top left key,
+and they'll progress in this order:
+
+1  2  3  4
+5  6  7  8
+
+Each key can do one of 5 different things.
+
+1. Single Key
+=============
+
+It can act as a single key, for example:
+
+>   ("Shoot", Keycode.ENTER),
+
+This will create a key called "Shoot" that acts exactly like
+the Enter key on a keyboard. You can hold the button down and
+get repeated presses, and when you release the button on the
+macropad, it'll stop pressing the Enter key.
```

Comment:
I've made this much clearer hopefully. Repeating a keystroke isn't currently supported on this one. It'd be a bit complex to add given the current architecture, but I can see it would be useful...